src/NuGet.Core/NuGet.Packaging.Core/PackageType.cs

```diff
@@ -38,6 +38,26 @@ public override bool Equals(object obj)
             return Equals(obj as PackageType);
         }
 
+        public static bool operator ==(PackageType a, PackageType b)
+        {
+            if (ReferenceEquals(a, b))
+            {
+                return true;
+            }
+            
+            if (((object)a == null) || ((object)b == null))
```

Comment:
Nope, since we cast to `object`, we use the `object` implementation of `==` operator, which only allows `null` to be equal to `null`.